casual/casual-jca/src/main/java/se/laz/casual/jca/CasualManagedConnection.java

```diff
@@ -84,20 +85,17 @@ public NetworkConnection getNetworkConnection()
         {
             if (networkConnection == null)
             {
-                Domain domain = ConfigurationService.getInstance().getConfiguration().getDomain();
-                NettyConnectionInformation ci = NettyConnectionInformation.createBuilder().withAddress(new InetSocketAddress(mcf.getHostName(), mcf.getPortNumber()))
-                                                                          .withProtocolVersion(mcf.getCasualProtocolVersion())
-                                                                          .withDomainId(domain.getId())
-                                                                          .withDomainName(domain.getName())
-                                                                          .build();
-                networkConnection = NettyNetworkConnection.of(ci, this);
-                log.finest(()->"created new nw connection " + this);
+                if(null != mcf.getNetworkPoolName() && null == mcf.getNetworkPoolSize())
+                {
+                    log.warning(() -> "networkPoolName set to: " + mcf.getNetworkPoolName() + " but missing networkPoolSize!");
+                }
+                networkConnection = null != mcf.getNetworkPoolSize() && null != mcf.getNetworkPoolName() ? NetworkPoolHandler.getInstance().getOrCreate( mcf.getNetworkPoolName(), mcf.getAddress(), mcf.getCasualProtocolVersion(), this, mcf.getNetworkPoolSize()) : createOneToOneManagedConnection();
```

Comment:
This line almost reaches the 300 character mark. Maybe reformat for readability. For a ternary operator with an assignment I would think something like the following would help for line length and clarity
```
var = <boolean expression>
        ? <value if true>
        : <value else>
```